hazelcast/src/main/java/com/hazelcast/agrona/collections/Int2ObjectHashMap.java

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014 Real Logic Ltd.
```

Comment:
Yes, whenever derivative work is added to a project, a `NOTICE` file is needed; this is certain. The only point I wasn't completely sure about is that copyright line. However, since this is truly _derivative_ work and not pure copy-paste (it backports Java 8 code to Java 6), it makes sense to me to have our copyright.